src/metrics/metrics.go

```diff
@@ -93,80 +117,146 @@ func populateWaitTimeMetrics(instanceEntity *integration.Entity, connection *con
 	}
 }
 
-func populateCustomMetrics(instanceEntity *integration.Entity, connection *connection.SQLConnection, query string) {
-
-	rows, err := connection.Queryx(query)
-	if err != nil {
-		log.Error("Could not execute custom query: %s", err.Error())
-		return
-	}
-
-	defer func() {
-		_ = rows.Close()
-	}()
+// Execute one or more custom queries
+func populateCustomMetrics(instanceEntity *integration.Entity, connection *connection.SQLConnection, queries []CustomQuery) {
+	// Execute each query in array
+	for _, query := range queries {
+		var prefix string
+		if len(query.Database) > 0 {
+			prefix = "USE "+query.Database+"; "
+		}
 
-	for rows.Next() {
-		row := make(map[string]interface{})
-		err := rows.MapScan(row)
+		rows, err := connection.Queryx(prefix + query.Query)
 		if err != nil {
-			log.Error("Failed to scan custom query row: %s", err)
+			log.Error("Could not execute custom query: %s", err.Error())
 			return
 		}
 
-		nameInterface, ok := row["metric_name"]
-		if !ok {
-			log.Error("Missing required column 'metric_name' in custom query")
-			return
-		}
-		name, ok := nameInterface.(string)
-		if !ok {
-			log.Error("Non-string type %T for custom query 'metric_name' column", nameInterface)
-			continue
-		}
+		defer func() {
+			_ = rows.Close()
+		}()
 
-		metricTypeInterface, ok := row["metric_type"]
-		if !ok {
-			log.Error("Missing required column 'metric_type' in custom query")
-			return
-		}
-		metricTypeString, ok := metricTypeInterface.(string)
-		if !ok {
-			log.Error("Non-string type %T for custom query 'metric_type' column", metricTypeInterface)
-			continue
-		}
-		metricType, err := metric.SourceTypeForName(metricTypeString)
-		if err != nil {
-			log.Error("Invalid metric type %s: %s", metricTypeString, err)
-			continue
-		}
+		for rows.Next() {
+			row := make(map[string]interface{})
+			err := rows.MapScan(row)
+			if err != nil {
+				log.Error("Failed to scan custom query row: %s", err)
+				return
```

Comment:
Now that we are doing multiple queries, we should probably `continue` after logging rather than `return` so that a single failure doesn't cause all their metrics to fail

Reply:
updated code